### ISSUE TITLE
Specifying directory trees other than $pwd

### ISSUE THREAD

**Issue by endophage** [GoASTScanner/gas#94]
It should be possible to provide `relative/or/absolute/path/...`. The ellipsis to indicate the parsing of a whole subtree should not be limited to `./...`. This is useful for CI systems so I don't necessarily have to change my working directory just to run gas.

**Comment by gcmurphy** [comment]
:+1: Been meaning to get to that for a while.

**Comment by endophage** [comment]
I'm happy to PR it in to you guys. I think it also necessitates a change to how you handle `-skip` though too, which for the moment I think makes some assumptions about the `pwd`.

I propose changing filelist to simply store the patterns provided to skip and directly applying `filepath.Match` rather than it's current building of a map via `filepath.Glob`.

If that's agreeable to you I can get it PR'd pretty quickly.

**Comment by gcmurphy** [comment]
Sounds great! Although fair warning getting filepath.Match to work correctly with path exclusion has been a source of many bugs / confusion in the past. However I will concede that filepath.Glob is a bandaid fix rather than a correct solution.

**Comment by endophage** [comment]
The problems with filepath.Match are possibly because as far as I can tell, the `**` syntax is not supported in Go :-/ I'd actually be surprised if the current implementation of fileList is working as desired.

c.f. 
  - Match rules: https://golang.org/pkg/path/filepath/#Match
  - Match code that essentially compacts multiple `*` on the front of the pattern into a single one https://golang.org/src/path/filepath/match.go#L92
  - Unresolved discussion on adding `**` support https://github.com/golang/go/issues/11862

**Comment by endophage** [comment]
Yeah, confirmed current fileList behaviour is not as desired. Rebuilt from master and added a single line to print `excluding: <path>` for each path captured by glob.

```
$ ./gas ./... 2>&1 | ag excluding
2016/12/02 10:44:18 excluding rules/bind_test.go
2016/12/02 10:44:18 excluding rules/blacklist_test.go
2016/12/02 10:44:18 excluding rules/errors_test.go
2016/12/02 10:44:18 excluding rules/fileperms_test.go
2016/12/02 10:44:18 excluding rules/hardcoded_credentials_test.go
2016/12/02 10:44:18 excluding rules/httpoxy_test.go
2016/12/02 10:44:18 excluding rules/nosec_test.go
2016/12/02 10:44:18 excluding rules/rand_test.go
2016/12/02 10:44:18 excluding rules/rsa_test.go
2016/12/02 10:44:18 excluding rules/sql_test.go
2016/12/02 10:44:18 excluding rules/subproc_test.go
2016/12/02 10:44:18 excluding rules/tempfiles_test.go
2016/12/02 10:44:18 excluding rules/templates_test.go
2016/12/02 10:44:18 excluding rules/tls_test.go
2016/12/02 10:44:18 excluding rules/unsafe_test.go
2016/12/02 10:44:18 excluding rules/utils_test.go
2016/12/02 10:44:18 excluding rules/weakcrypto_test.go
$ ls
LICENSE.txt     core            gas             output          rules           tools_test.go
README.md       filelist.go     main.go         rulelist.go     tools.go
```

You can see in the `ls` there's a `tools_test.go` I added that should be captured by the globbing, but it's considering `**/*_test.go` to be `*/*_test.go` and only capturing paths with exactly one directory. I also tried with a `*_test.go` file 2 levels deep and that also wasn't matched by `Glob`

**Comment by endophage** [comment]
How would you feel about using [this](https://github.com/ryanuber/go-glob) and just having simple globs?

**Comment by gcmurphy** [comment]
I think something like that would be ok as it's fairly minimal.

**Comment by endophage** [comment]
Great, will open the PR now but still writing some tests so not quite ready to merge!

**Comment by endophage** [comment]
Closed by #98
